src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ValueProviderResult.cs

```diff
@@ -142,9 +142,17 @@ private object ConvertSimpleType(CultureInfo culture, object value, Type destina
             if (!(canConvertFrom || converter.CanConvertTo(destinationType)))
             {
                 // EnumConverter cannot convert integer, so we verify manually
-                if (destinationType.GetTypeInfo().IsEnum && (value is int))
+                if (destinationType.GetTypeInfo().IsEnum &&
+                    (value is int ||
+                    value is uint ||
+                    value is long ||
+                    value is ulong ||
+                    value is short ||
+                    value is ushort ||
+                    value is byte ||
+                    value is sbyte))
                 {
-                    return Enum.ToObject(destinationType, (int)value);
+                    return Enum.ToObject(destinationType, Convert.ToInt32(value));
```

Comment:
W.r.t. using `Enum.ToObject()`, note the rest of this line :smiley_cat:  The cast or conversion of this argument is not needed at all.